class A {
    fun bar() {
        <!NONE_APPLICABLE!>div<!> {
            <!NONE_APPLICABLE!>div<!> {
                <!NONE_APPLICABLE!>div<!> {
                    <!NONE_APPLICABLE!>div<!> {
                        <!NONE_APPLICABLE!>div<!> {
                            <!NONE_APPLICABLE!>div<!> {
                                <!NONE_APPLICABLE!>div<!> {
                                    <!NONE_APPLICABLE!>div<!> {
                                        <!NONE_APPLICABLE!>div<!> {
                                            <!NONE_APPLICABLE!>div<!> {
                                                <!NONE_APPLICABLE!>div<!> {
                                                    <!NONE_APPLICABLE!>div<!> {
                                                        <!NONE_APPLICABLE!>div<!> {
                                                            <!NONE_APPLICABLE!>div<!> {
                                                                <!NONE_APPLICABLE!>div<!> {
                                                                    <!NONE_APPLICABLE!>div<!> {
                                                                        <!NONE_APPLICABLE!>div<!> {
                                                                            <!NONE_APPLICABLE!>div<!> {
                                                                                <!NONE_APPLICABLE!>div<!> {
                                                                                    <!NONE_APPLICABLE!>div<!> {
                                                                                        <!NONE_APPLICABLE!>div<!> {
                                                                                            <!NONE_APPLICABLE!>div<!> {
                                                                                                <!NONE_APPLICABLE!>div<!> {
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }

                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
